lib/js/src/protocol/SdlProtocolListener.js

```diff
@@ -162,6 +162,45 @@ class SdlProtocolListener {
             return this._getSessionId();
         }
     }
+
+
+    /**
+     * Sets the getDesiredVideoParams getter
+     * @param {function} getter
+     */
+    setGetDesiredVideoParams (getter) {
+        this._getDesiredVideoParams = getter;
+    }
+
+    /**
+     * Invoke the getDesiredVideoParams getter
+     * @returns {VideoStreamingParameters} 
+     */
+    getDesiredVideoParams () {
+        if (typeof this._getDesiredVideoParams === 'function') {
+            return this._getDesiredVideoParams();
+        }
+    }
+
+    /**
+     * Sets the setSetAcceptedVideoParams setter
+     * @param {function} setter 
+     */
+    setSetAcceptedVideoParams (setter) {
+        this._setAcceptedVideoParams = setter;
+    }
+
+
+    /**
+     * Invoke the setAcceptedVideoParams setter
+     * @param {VideoStreamingParameters} params 
+     */
+    setAcceptedVideoParams (params) {
+        if (typeof this._setAcceptedVideoParams === 'function') {
+            this._setAcceptedVideoParams(params);
+            return this;
```

Comment:
Return should happen outside of the scope of the if() statement to ensure `this` is always returned.